web/src/custom_profile_fields_ui.ts

```diff
@@ -45,18 +46,41 @@ export function append_custom_profile_fields(element_id: string, user_id: number
             rendered_value: "",
         };
         const editable_by_user = current_user.is_admin || field.editable_by_user;
-        const is_select_field = field.type === all_field_types.SELECT.id;
+        const is_select_field =
+            field.type === all_field_types.SELECT.id ||
+            field.type === all_field_types.SELECT_MULTIPLE.id;
         const field_choices = [];
 
         if (is_select_field) {
-            const field_choice_dict = settings_components.select_field_data_schema.parse(
-                JSON.parse(field.field_data),
-            );
+            let field_choice_dict: Record<string, {text: string; order: string}> = {};
+            try {
+                field_choice_dict = settings_components.select_field_data_schema.parse(
+                    JSON.parse(field.field_data),
+                );
+            } catch {
+                field_choice_dict = {};
+            }
+            const current_val = field_value.value;
+            const selected_values = new Set<string>();
+            if (field.type === all_field_types.SELECT_MULTIPLE.id) {
+                let parsed: unknown;
+                try {
+                    parsed = JSON.parse(current_val);
+                } catch {
+                    parsed = [];
+                }
+                const result = z.array(z.string()).safeParse(parsed);
+                if (result.success) {
+                    for (const item of result.data) {
+                        selected_values.add(item);
+                    }
+                }
+            }
```

Comment:
This is a lot of parsing code; can you add comments explaining what possible failures you're worried about here?